### PR TITLE
Start adding user email support

### PR DESCRIPTION
**Pull request by migurski**
Adds an email address to the profile form, and sends emails to one of our five pre-selected addresses when an address is added.

- [x] Add configuration for email sending
- [x] Add confirmation emails when user changes their address

Not yet:

- Send alert emails (SMS ones don't exist at all yet)
- PIN numbers other than "1234"

Part of #59.

### CODE REVIEW

**Comment by migurski** [comment]
I've created a VLG Mailgun account. It’s currently free which can only send email to specified addresses. We'll need to upgrade it in order to send emails to other users per these requirements.

File: CDTADPQ/web/__init__.py [hunk 1]
```diff
@@ -115,22 +121,72 @@ def get_zipcode():
...
+    email_addr_str = email_address or ''
     return flask.render_template('profile.html', phone_number=phone_number,
-                                 zip_codes=zip_code_str, **template_kwargs())
+                                 zip_codes=zip_code_str, email_address=email_addr_str,
+                                 **template_kwargs())
+
+@app.route('/profile', methods=['POST'])
+@user_is_logged_in
+def post_profile():
+    print('FUCK', flask.request.form)
```
Comment (pui) [inline CDTADPQ/web/__init__.py]: Let's not commit swear words to the repo!

Comment (pui) [inline CDTADPQ/web/__init__.py]: Erp, wasn't super paying attention that this is still in progress.

Comment (migurski) [inline CDTADPQ/web/__init__.py]: 👍

**Comment by migurski** [comment]
Swear word removed.

**Comment by migurski** [comment]
Not in progress by the way; I would like a merge on this.

**Comment by phae** [comment]
I got a 500 error when I tried to add my email address with a + in it, then I went back and tried again without a + pattern and it submitted. Just waiting for my email code now. :)

**Comment by phae** [comment]
Changing my email address gave me a 500 (no sneaky +).

**Comment by migurski** [comment]
That's probably due to our free Mailgun plan allowing only a short whitelist of emails.

File: CDTADPQ/web/__init__.py [hunk 2]
```diff
@@ -115,22 +121,72 @@ def get_zipcode():
...
+    redirect_url = flask.url_for('get_email_addressed', address_encoded=address_encoded)
+
+    return flask.redirect(redirect_url, code=303)
+
+@app.route('/profile/email-addressed/<address_encoded>', methods=['GET'])
+@user_is_logged_in
+def get_email_addressed(address_encoded):
+    signer = itsdangerous.URLSafeSerializer(flask.current_app.secret_key)
+    email_address = signer.loads(address_encoded)
+    print('POOP', email_address, 'from', address_encoded)
```
Comment (pui) [inline CDTADPQ/web/__init__.py]: Just a reminder to remove 💩  as well

**Comment by pui** [comment]
Looks good from the code side of things

**Comment by migurski** [comment]
Great, thanks!

**Pull request merged**